staging/src/k8s.io/client-go/examples/third-party-resources/main.go

```diff
@@ -52,66 +53,65 @@ func main() {
 	}
 
 	// initialize third party resource if it does not exist
-	tpr, err := clientset.ExtensionsV1beta1().ThirdPartyResources().Get("example.k8s.io", metav1.GetOptions{})
-	if err != nil {
-		if errors.IsNotFound(err) {
-			tpr := &v1beta1.ThirdPartyResource{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "example.k8s.io",
-				},
-				Versions: []v1beta1.APIVersion{
-					{Name: "v1"},
-				},
-				Description: "An Example ThirdPartyResource",
-			}
-
-			result, err := clientset.ExtensionsV1beta1().ThirdPartyResources().Create(tpr)
-			if err != nil {
-				panic(err)
-			}
-			fmt.Printf("CREATED: %#v\nFROM: %#v\n", result, tpr)
-		} else {
-			panic(err)
-		}
-	} else {
-		fmt.Printf("SKIPPING: already exists %#v\n", tpr)
+	err = exampleclient.CreateTPR(clientset)
+	if err == nil {
+		// See the issue https://github.com/kubernetes/features/issues/95
+		// ("Make new TPRs available immediately after the create request succeeds")
+		fmt.Print("Sleeping to make sure the TPR is processed and available\n")
+		time.Sleep(5 * time.Second)
+	} else if !apierrors.IsAlreadyExists(err) {
+		panic(err)
 	}
 
 	// make a new config for our extension's API group, using the first config as a baseline
-	var tprconfig *rest.Config
-	tprconfig = config
-	configureClient(tprconfig)
-
-	tprclient, err := rest.RESTClientFor(tprconfig)
+	exampleClient, _, err := exampleclient.NewClient(config)
 	if err != nil {
 		panic(err)
 	}
 
-	var example Example
+	// start a watcher on instances of our TPR
+	watcher := ExampleController{
+		clientset:     clientset,
+		exampleClient: exampleClient,
+	}
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+	go watcher.Run(ctx)
 
-	err = tprclient.Get().
-		Resource("examples").
-		Namespace(v1.NamespaceDefault).
+	// The sleep below is just to make sure that the watcher.Run() goroutine has successfully executed
+	// and the watcher is handling the events about Example TPR instances.
+	// In the normal application there is no need for it, because:
+	// 1. It's unlikely to create a watcher and a TPR instance at the same time in the same application.
+	// 2. The application with watcher would most probably keep running instead of exiting right after the watcher startup.
+	time.Sleep(5 * time.Second)
+
+	// GET/POST an instance of our TPR
+	var example tprv1.Example
+
+	err = exampleClient.Get().
+		Resource(tprv1.ExampleResourcePlural).
+		Namespace(apiv1.NamespaceDefault).
 		Name("example1").
 		Do().Into(&example)
 
 	if err != nil {
-		if errors.IsNotFound(err) {
+		if apierrors.IsNotFound(err) {
```

Comment:
Just try to post the tpr and check for errors.IsAlreadyExists. No need for the whole Get+Post logic here.